chia/wallet/dao_wallet/dao_wallet.py

```diff
@@ -0,0 +1,2268 @@
+        amounts: List[uint64],
+        asset_types: List[Optional[bytes32]] = [None],
+    ) -> Program:
+        if len(recipient_puzhashes) != len(amounts) != len(asset_types):  # pragma: no cover
+            raise ValueError("Mismatch in the number of recipients, amounts, or asset types")
+        xch_conds = []
+        cat_conds = []
+        for recipient_puzhash, amount, asset_type in zip(recipient_puzhashes, amounts, asset_types):
+            if asset_type:
+                cat_conds.append([asset_type, [[51, recipient_puzhash, amount]]])
+            else:
+                xch_conds.append([51, recipient_puzhash, amount])
+        puzzle = get_spend_p2_singleton_puzzle(self.dao_info.treasury_id, Program.to(xch_conds), Program.to(cat_conds))
+        return puzzle
+
+    async def generate_update_proposal_innerpuz(
+        self,
+        new_dao_rules: DAORules,
+        new_proposal_validator: Optional[Program] = None,
+    ) -> Program:
+        if not new_proposal_validator:
+            assert isinstance(self.dao_info.current_treasury_innerpuz, Program)
+            new_proposal_validator = get_proposal_validator(
+                self.dao_info.current_treasury_innerpuz, new_dao_rules.proposal_minimum_amount
+            )
+        return get_update_proposal_puzzle(new_dao_rules, new_proposal_validator)
+
+    async def generate_mint_proposal_innerpuz(
+        self,
+        amount_of_cats_to_create: uint64,
+        cats_new_innerpuzhash: bytes32,
+    ) -> Program:
+        if amount_of_cats_to_create % 2 == 1:  # pragma: no cover
+            raise ValueError("Minting proposals must mint an even number of CATs")
+        cat_launcher = create_cat_launcher_for_singleton_id(self.dao_info.treasury_id)
+
+        cat_wallet: CATWallet = self.wallet_state_manager.wallets[self.dao_info.cat_wallet_id]
+        cat_tail_hash = cat_wallet.cat_info.limitations_program_hash
+        eve_puz_hash = curry_cat_eve(cats_new_innerpuzhash)
+        full_puz = construct_cat_puzzle(CAT_MOD, cat_tail_hash, eve_puz_hash)
+        xch_conditions = [
+            [
+                51,
+                cat_launcher.get_tree_hash(),
+                uint64(amount_of_cats_to_create),
+                [cats_new_innerpuzhash],
+            ],  # create cat_launcher coin
+            [
+                60,
+                Program.to([ProposalType.MINT.value, full_puz.get_tree_hash()]).get_tree_hash(),
+            ],  # make an announcement for the launcher to assert
+        ]
+        puzzle = get_spend_p2_singleton_puzzle(self.dao_info.treasury_id, Program.to(xch_conditions), [])
+        return puzzle
+
+    async def generate_new_proposal(
+        self,
+        proposed_puzzle: Program,
+        tx_config: TXConfig,
+        vote_amount: Optional[uint64] = None,
+        fee: uint64 = uint64(0),
+        push: bool = True,
+        extra_conditions: Tuple[Condition, ...] = tuple(),
+    ) -> Union[SpendBundle, TransactionRecord]:
```

Comment:
Same comments as in the dao_cat_wallet file about transaction methods only returning `List[TransactionRecord]` and not pushing them automatically.

Reply:
done